src/BugsnagServiceProvider.php

```diff
@@ -180,6 +180,71 @@ protected function setupQueue(QueueManager $queue)
 
             $this->app->make(Tracker::class)->set($job);
         });
+
+        $this->setupQueueForLaravelVapor($queue);
+    }
+
+    /**
+     * Setup queue events for Laravel Vapor.
+     *
+     * This is required because Laravel Vapor's queue system doesn't behave as
+     * a daemonised queue worker (the 'looping' event never fires) but also
+     * doesn't behave as a non-daemonised queue worker (our shutdown function
+     * never fires).
+     *
+     * @param QueueManager $queue
+     *
+     * @return void
+     */
+    private function setupQueueForLaravelVapor(QueueManager $queue)
+    {
+        // ensure we're running on vapor
+        // this is how vapor-core does it, e.g.:
+        // https://github.com/laravel/vapor-core/blob/61437221090850ba6e51dce15d0058d362654f9b/src/ConfiguresAssets.php#L16-L19
+        if (!isset($_ENV['VAPOR_SSM_PATH'])) {
+            return;
+        }
+
+        // used to keep track of if we're the ones disabling batch sending, so we
+        // know if we need to re-enable it - if the user disables batch sending
+        // then they don't want it enabled at all
+        static $batchSendingWasDisabledByUs = false;
+
+        $queue->before(function () use (&$batchSendingWasDisabledByUs) {
+            // clear breadcrumbs to stop them leaking between jobs
+            $this->app->bugsnag->clearBreadcrumbs();
+
+            // only re-enable batch sending if we're the ones disabling it
+            // this allows users to disable batch sending entirely
+            if ($batchSendingWasDisabledByUs) {
+                $this->app->bugsnag->setBatchSending(true);
```

Comment:
Is there any potential impact to the customer part of the app if we temporarily disable batching? e.g. if they have an `after` or `exceptionOccurred` listener that relies on it and is executed before ours?

I wonder if it could be worth noting in the docs that it temporarily disables batching to ensure delivery?

Reply:
I'm not sure it's possible to rely on batch sending being enabled/disabled in any way; it's really just a performance optimisation

Maybe I'm misunderstanding what you're saying

> if they have an after or exceptionOccurred listener that relies on it and is executed before ours?

If their after or exceptionOccurred listener fires before ours, batch sending would be enabled and the `flush` would send events. If their listener fires after ours, batch sending would be disabled and the events would send immediately

Our listener _should_ always fire first because our service provider should be registered before any of the user's providers

Reply:
Yes, I was thinking of the case where they have a listener after ours that for some reason is expecting batching to be on. They would see an unexpected state where it temporarily wasn't. However I cannot imagine a reason why Bugsnag batch mode would be something referenced in a user callback, it was more a technical possibility that occurred to me.